diff -Naur enigma2-plugin-youtube/src/YouTubeSearch.py enigma2-plugin-youtube-PersianPrince/src/YouTubeSearch.py
--- enigma2-plugin-youtube/src/YouTubeSearch.py	2021-01-29 12:38:25.000000000 +0330
+++ enigma2-plugin-youtube-PersianPrince/src/YouTubeSearch.py	2021-01-29 12:46:56.058419764 +0330
@@ -2,7 +2,7 @@
 
 from threading import Thread
 from json import loads
-
+import os
 from enigma import ePoint, getDesktop
 from Screens.ChoiceBox import ChoiceBox
 from Screens.Screen import Screen
@@ -15,7 +15,7 @@
 from Components.Sources.Boolean import Boolean
 from Components.Sources.List import List
 from Components.Sources.StaticText import StaticText
-
+from Tools.Directories import resolveFilename, SCOPE_PLUGINS
 from . import _
 from .compat import compat_quote
 from .compat import compat_ssl_urlopen
@@ -312,8 +312,20 @@
 
 	def openKeyboard(self):
 		self['config'].getCurrent()[1].help_window.instance.hide()
-		self.session.openWithCallback(self.keyBoardCallback, YouTubeVirtualKeyBoard,
-			text=self.searchValue.value)
+		self.VirtualKeyBoard = config.plugins.YouTube.VirtualKeyBoard.value
+		if self.VirtualKeyBoard == "YouTube":
+			if os.path.islink("/usr/lib/enigma2/python/Screens/VirtualKeyBoard.py") and os.path.exists(resolveFilename(SCOPE_PLUGINS, "SystemPlugins/NewVirtualKeyBoard")):
+				from Screens.VirtualKeyBoard import VirtualKeyBoard
+				self.session.openWithCallback(self.keyBoardCallback, VirtualKeyBoard, '')
+			else:
+				self.session.openWithCallback(self.keyBoardCallback, YouTubeVirtualKeyBoard,
+					text = self.searchValue.value)
+		elif self.VirtualKeyBoard == "Image":
+			from Screens.VirtualKeyBoard import VirtualKeyBoard
+			self.session.openWithCallback(self.keyBoardCallback, VirtualKeyBoard, '')
+		else:
+			self.session.openWithCallback(self.keyBoardCallback, YouTubeVirtualKeyBoard,
+				text = self.searchValue.value)
 
 	def keyBoardCallback(self, name):
 		config = self['config'].getCurrent()[1]
diff -Naur enigma2-plugin-youtube/src/YouTubeUi.py enigma2-plugin-youtube-PersianPrince/src/YouTubeUi.py
--- enigma2-plugin-youtube/src/YouTubeUi.py	2021-01-29 12:38:25.000000000 +0330
+++ enigma2-plugin-youtube-PersianPrince/src/YouTubeUi.py	2021-01-29 12:48:25.002631989 +0330
@@ -70,6 +70,8 @@
 	('playprev', _('Play previous'))])
 config.plugins.YouTube.onMovieStop = ConfigSelection(default='ask', choices=[
 	('ask', _('Ask user')), ('quit', _('Return to list'))])
+config.plugins.YouTube.VirtualKeyBoard = ConfigSelection(default='YouTube', choices=[
+	('YouTube', _('YouTube VirtualKeyBoard')), ('Image', _('Image VirtualKeyBoard'))])
 config.plugins.YouTube.login = ConfigYesNo(default=False)
 config.plugins.YouTube.downloadDir = ConfigDirectory(default=resolveFilename(SCOPE_HDD))
 config.plugins.YouTube.mergeFiles = ConfigYesNo(default=False)
@@ -1404,6 +1406,9 @@
 		configlist.append(getConfigListEntry(_('Merge downloaded files:'),
 			config.plugins.YouTube.mergeFiles,
 			_('FFmpeg will be used to merge downloaded DASH video and audio files.\nFFmpeg will be installed if necessary.')))
+		configlist.append(getConfigListEntry(_('Choose VirtualKeyBoard style:'),
+			config.plugins.YouTube.VirtualKeyBoard,
+			_('You can choose what style of VirtualKeyBoard to use.\nYouTube or Image (VirtualKeyBoard).')))
 		for p in plugins.getPlugins(where=PluginDescriptor.WHERE_MENU):
 			# TRANSLATORS: Don't translate this! It is used as a variable, so it must be equal to the translation in the plugin!
 			if p.name == _("ServiceApp"):
